### PR TITLE
Implemented Date Time Duration type #99

### PR DESCRIPTION
**Pull request by emir-sehovic1**
- Created a Date Time Duration value object with TimeSpan as underlying type
- Date Time Duration can be created from native TimeSpan type or from various options by specifying days, hours, minutes, seconds or milliseconds

### CODE REVIEW

File: tests/Nox.Types.Tests/Types/DateTimeDuration/DateTimeDurationTests.cs [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    }
+
+    [Fact]
+    public void From_WithNegativeTimeSpanInput_ReturnsAbsoluteValue()
+    {
+        var timeSpan = new TimeSpan(1, 2, 3, 4, 5);
+
+        var dateTimeDuration = DateTimeDuration.From(-timeSpan);
+
+        dateTimeDuration.Value.Should().Be(timeSpan.Duration());
```
Comment (dejobratic) [inline tests/Nox.Types.Tests/Types/DateTimeDuration/DateTimeDurationTests.cs:26]: Can we have a hardcoded expected value instead of timeSpan.Duration()?

**Review by dejobratic — CHANGES_REQUESTED** [review]
Good job. Just a minor comments on tests.

File: tests/Nox.Types.Tests/Types/DateTimeDuration/DateTimeDurationTests.cs (hunk 1 above)
Comment (emir-sehovic1) [reply tests/Nox.Types.Tests/Types/DateTimeDuration/DateTimeDurationTests.cs]: Thanks for the suggestion, I've updated the tests

**Review by dejobratic — APPROVED** [review]

File: src/Nox.Types/Types/DateTimeDuration/DateTimeDuration.cs
Comment (rochar) [inline src/Nox.Types/Types/DateTimeDuration/DateTimeDuration.cs:1]: We need to think if we need a TypesOptions, for example Max/Min Duration

**Review by rochar — CHANGES_REQUESTED** [review]
looks good small comment

**Comment by rochar** [comment]
please also update to main

File: src/Nox.Types/Types/DateTimeDuration/DateTimeDuration.cs
Comment (emir-sehovic1) [reply src/Nox.Types/Types/DateTimeDuration/DateTimeDuration.cs:1]: Thanks for the suggestion, I've added the duration options

File: src/Nox.Types/Types/DateTimeDuration/DateTimeDurationTypeOptions.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public static readonly int DefaultMinDurationMinutes = 0;
+    public static readonly int DefaultMinDurationSeconds = 0;
+    public static readonly int DefaultMinDurationMilliseconds = 0;
+
+    public int MaxDurationDays { get; set; } = DefaultMaxDurationDays;
+    public int MaxDurationHours { get; set; } = DefaultMaxDurationHours;
+    public int MaxDurationMinutes { get; set; } = DefaultMaxDurationMinutes;
+    public int MaxDurationSeconds { get; set; } = DefaultMaxDurationSeconds;
+    public int MaxDurationMilliseconds { get; set; } = DefaultMaxDurationMilliseconds;
+    public TimeSpan GetMaxDuration() => new TimeSpan(MaxDurationDays, MaxDurationHours, MaxDurationMinutes, MaxDurationSeconds, MaxDurationMilliseconds).Duration();
```
Comment (rochar) [inline src/Nox.Types/Types/DateTimeDuration/DateTimeDurationTypeOptions.cs:24]: TypeOptions looks good. My expectation to GetMaxDuration was to use a single field to define the max timespan, for example TimeSpan.FromMilliseconds()... From a user perspective I would argue that the expectation would be to create a MaxTimeSpan from a unit... and not set all this values, to mee seems complex..

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, just a small topic to discuss

**Review by rochar — APPROVED** [review]

**Comment by rochar** [comment]
fix the conflicts, update with main and you can merge. Thks

**Pull request merged**